discovery/eureka/client_test.go

```diff
@@ -0,0 +1,213 @@
+        <serviceUpTimestamp>1535444352472</serviceUpTimestamp>
+      </leaseInfo>
+      <metadata>
+        <project>meta-service</project>
+        <management.port>8090</management.port>
+      </metadata>
+      <homePageUrl>http://meta-service002.test.com:8080/</homePageUrl>
+      <statusPageUrl>http://meta-service002.test.com:8080/info</statusPageUrl>
+      <healthCheckUrl>http://meta-service002.test.com:8080/health</healthCheckUrl>
+      <vipAddress>meta-service</vipAddress>
+      <secureVipAddress>meta-service</secureVipAddress>
+      <isCoordinatingDiscoveryServer>false</isCoordinatingDiscoveryServer>
+      <lastUpdatedTimestamp>1535444352472</lastUpdatedTimestamp>
+      <lastDirtyTimestamp>1535444352398</lastDirtyTimestamp>
+      <actionType>ADDED</actionType>
+    </instance>
+    <instance>
+      <instanceId>meta-service001.test.com:meta-service:8080</instanceId>
+      <hostName>meta-service001.test.com</hostName>
+      <app>META-SERVICE</app>
+      <ipAddr>192.133.87.236</ipAddr>
+      <status>UP</status>
+      <overriddenstatus>UNKNOWN</overriddenstatus>
+      <port enabled="true">8080</port>
+      <securePort enabled="false">443</securePort>
+      <countryId>1</countryId>
+      <dataCenterInfo class="com.netflix.appinfo.InstanceInfo$DefaultDataCenterInfo">
+        <name>MyOwn</name>
+      </dataCenterInfo>
+      <leaseInfo>
+        <renewalIntervalInSecs>30</renewalIntervalInSecs>
+        <durationInSecs>90</durationInSecs>
+        <registrationTimestamp>1535444352472</registrationTimestamp>
+        <lastRenewalTimestamp>1596110168846</lastRenewalTimestamp>
+        <evictionTimestamp>0</evictionTimestamp>
+        <serviceUpTimestamp>1535444352472</serviceUpTimestamp>
+      </leaseInfo>
+      <metadata>
+        <project>meta-service</project>
+        <management.port>8090</management.port>
+      </metadata>
+      <homePageUrl>http://meta-service001.test.com:8080/</homePageUrl>
+      <statusPageUrl>http://meta-service001.test.com:8080/info</statusPageUrl>
+      <healthCheckUrl>http://meta-service001.test.com:8080/health</healthCheckUrl>
+      <vipAddress>meta-service</vipAddress>
+      <secureVipAddress>meta-service</secureVipAddress>
+      <isCoordinatingDiscoveryServer>false</isCoordinatingDiscoveryServer>
+      <lastUpdatedTimestamp>1535444352472</lastUpdatedTimestamp>
+      <lastDirtyTimestamp>1535444352398</lastDirtyTimestamp>
+      <actionType>ADDED</actionType>
+    </instance>
+  </application>
+</applications>`
+
+	// Simulate apps with a valid XML response.
+	respHandler := func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		w.Header().Set("Content-Type", "application/xml")
+		io.WriteString(w, appsXML)
+	}
+	// Create a test server with mock HTTP handler.
+	ts := httptest.NewServer(http.HandlerFunc(respHandler))
+	defer ts.Close()
+
+	apps, err := fetchApps(context.TODO(), []string{ts.URL}, &http.Client{})
+	if err != nil {
```

Comment:
you can use `testutil.Ok(t, err)` to fail if an error was returned

Reply:
I'll change it to testutil.